tests/serverpod_test_server/test_integration/serverpod_cli_args_override_test.dart

```diff
@@ -0,0 +1,342 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_shared/serverpod_shared.dart';
+import 'package:serverpod_test_server/src/generated/endpoints.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+void main() {
+  // These tests verify that command line arguments correctly override
+  // ServerpodConfig values when both are provided to the Serverpod constructor.
+  // Note: runMode has special handling where it's calculated from CLI args first,
+  // then environment variables, then defaults to 'development'.
```

Comment:
Suggestion: Remove this description; this type of information should be embedded in the test description/structure.